erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -1590,3 +1590,33 @@ def create_invoice_discounting(source_name, target_doc=None):
 def get_item_details(item_code):
 	item = frappe.get_doc("Item", item_code).as_dict()
 	return item
+
+@frappe.whitelist()
+def email_coas(docname):
+	sales_invoice = frappe.get_doc("Sales Invoice", docname).as_dict()
+
+	for item in sales_invoice.get("items"):
+		delivery_note = frappe.get_doc("Delivery Note", item.delivery_note).as_dict()
+
+		if not delivery_note.get("contact_email"):
+			frappe.throw(_("No contact email found"))
+
+		coas = item.get("certificate_of_analysis")
+
+		if not coas:
+			frappe.msgprint(_("No Certificates of Analysis attached"))
+			return
+
+		attachments = []
```

Comment:
```suggestion
		attachments = []
```

- Only attachments for last item will be sent.
- move attachments list out of for loop